File: api/src/models/team.ts
import { z } from 'zod';
import { baseEntitySchema } from './base';

export const teamPlayerSchema = z.object({
  playerId: z.string(),
  role: z.enum(['captain', 'player']),
  joinedAt: z.string(),
});

export type TeamPlayer = z.infer<typeof teamPlayerSchema>;

export const teamSchema = baseEntitySchema.extend({
  name: z.string(),
  captainId: z.string(),
  city: z.string().optional(),
  homeVenue: z.string().optional(),
  logoUrl: z.string().optional(),
  members: z.array(teamPlayerSchema),
});

export type Team = z.infer<typeof teamSchema>;

File: PoolDN/PoolDN/Models/Team.swift
import Foundation

struct TeamMember: Codable, Hashable {
    let playerId: String
    let role: String
    let joinedAt: String
    let name: String?
    let nickname: String?
    let avatarUrl: String?
}

struct Team: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let captainId: String
    let city: String?
    let homeVenue: String?
    let logoUrl: String?
    let members: [TeamMember]
    let createdAt: Int
    let updatedAt: Int
}
